src/core.c/Exception.pm6

```diff
@@ -3184,10 +3184,11 @@ my class X::MultipleTypeSmiley does X::Comp {
 }
 
 my class X::Seq::Consumed is Exception {
+    has Str:D $.kind = "Seq";
```

Comment:
I think it'd be more appropriate to pass the type object or the `Sequence` itself into the exception class, and leaving `message` to do the stringification.

Reply:
It was the initial plan, but then... Let's put it that way: I was a bit tired and for some reason decided `Seq` is not available here.